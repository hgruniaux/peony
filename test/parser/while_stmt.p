//# RUN: peony %s

fn test() {
    while true {}

    while 5i32 {}
    //~^ ERROR: mismatched types; expected 'bool', found 'i32'

    while () {}
    //~^ ERROR: expected expression
}
