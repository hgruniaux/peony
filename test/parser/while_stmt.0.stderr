<source>:6:11: error: mismatched types; expected 'bool', found 'i32'
    6 |     while 5i32 {}
      |           ~~~~
<source>:9:12: error: expected expression
    9 |     while () {}
      |            ^
